runtime/jvmti/suspendhelper.cpp

```diff
@@ -96,8 +96,13 @@ suspendThread(J9VMThread *currentThread, jthread thread, BOOLEAN allowNull, BOOL
 		if (VM_VMHelpers::isThreadSuspended(currentThread, threadObject)) {
 			rc = JVMTI_ERROR_THREAD_SUSPENDED;
 		} else {
-			U_64 internalSuspendState = (U_64)VM_VMHelpers::getCarrierVMThread(currentThread, threadObject);
-			J9OBJECT_U64_STORE(currentThread, threadObject, vm->internalSuspendStateOffset, (internalSuspendState | J9_VIRTUALTHREAD_INTERNAL_STATE_SUSPENDED));
+			J9VMThread *carrierVMThread = VM_VMHelpers::getCarrierVMThread(currentThread, threadObject);
+			J9OBJECT_U64_STORE(currentThread, threadObject, vm->internalSuspendStateOffset, ((U_64)carrierVMThread | J9_VIRTUALTHREAD_INTERNAL_STATE_SUSPENDED));
+			if (NULL != carrierVMThread) {
+				/* Decrement inspector count to reflect vthread suspended in transition. */
+				MM_ObjectAccessBarrierAPI objectAccessBarrier = MM_ObjectAccessBarrierAPI(currentThread);
+				objectAccessBarrier.inlineMixedObjectCompareAndSwapU64(currentThread, threadObject, vm->virtualThreadInspectorCountOffset, (U_64)-1, (U_64)-2);
+			}
```

Comment:
-1 should only be encountered for the SuspendAll and ResumeAll cases. Normal Suspend and Resume won't encounter -1 since they use getVMThread instead of exclusive VM access. can we remove this CAS?